src/explore-education-statistics-common/src/validation/serverValidations.ts

```diff
@@ -163,13 +163,22 @@ export function isServerValidationError(
 
 export function hasServerValidationError(
   error: Error,
-  errorMessage: string,
+  ...errorCodes: string[]
 ): boolean {
   if (isServerValidationError(error) && error.response?.data) {
-    const errors = Object.values(error.response?.data.errors);
-    if (errors.flat().includes(errorMessage)) {
-      return true;
-    }
+    const existingCodes = Object.values(error.response?.data.errors).flat();
+    return existingCodes.every(existingCode =>
+      errorCodes.includes(existingCode),
+    );
   }
   return false;
 }
+
+export function getServerValidationError(error: Error): string | null {
```

Comment:
Would suggest the following changes to this method:

```ts
export function getServerValidationErrorMessage<T extends string = string>(
  error: AxiosError<ServerValidationErrorResponse>, 
  fieldName = ''
): T | undefined
```

Note the following:
- A type parameter `T` has been added to let us specify the return type for string literal (instead of using `as` to coerce the type)
- The error's type has been changed to `AxiosError<ServerValidationErrorResponse>`. The intention would be to use `isServerValidationError` beforehand to assert the type in calling code. Alternatively, we could use `error: unknown`, but it could be a bit wasteful as we then potentially have to run `isServerValidationError` multiple times.
- We specify the field name to look for the error message under (see below).
- We return undefined as this is a bit terser in combination with the comment below around looking up using a field name.

Reply:
Nice improvements, and done, thanks.